### DISCUSSION TITLE
Error during backup: internal error: bitmap 'virtnbdbackup.7' not found in backing chain of 'vda']

### DISCUSSION THREAD

**Discussion by rettw78** [abbbi/virtnbdbackup#119] [Q&A / Support] [answered]
Hello:

I have virtnbdbackup 1.9.25 loaded on a Linux KVM/QEMU/libvirt server to backup by domains.  I cannot seem to get one of the domains to start a new backup chain in an empty folder - I get this error:

ERROR root checkpoint - delete [MainThread]: Error during checkpoint removal: [internal error: bitmap 'virtnbdbackup.7' not found in backing chain of 'vda']

I removed all bitmaps with qemu-img tool as part of the troubleshooting, trying to get this one VM to start a fresh set of backups in a new folder, and I can't seem to figure out how to get past this error.

Let me know what other information may be helpful in....

Rett Walters

**Comment by abbbi** [comment]
are the virtual machines moving between libvirt hosts? Please check:

https://github.com/abbbi/virtnbdbackup#transient-virtual-machines-checkpoint-persistency

Or was the virtual machine restored once? then check:

https://github.com/abbbi/virtnbdbackup/tree/dev#post-restore-steps-and-considerations

it seems there is an situation in your environment where an inconsistency between the checkpoints saved by libvirt and the bitmaps stored in the qemu images happens. What happens in detail is that during a full backup virtnbdbackup wants to remove all the checkpoints that are existant using libvirt API call. But libvirt is unable to find the bitmaps in the qemu image files, so the removal fails.

You can check with "virsh checkpoint-list [domain]" which checkpoints libvirt thinks are existant and use checkpoint-delete with the --metadata option to remove any dangling entries where the bitmap referenced by a checkpoint doesnt exist anymore.

Removing the bitmaps manually via qemu-img is never an good option. In the first place you have to check why this situation appeared for this domain. If the domain was restored prior to the backup, please check: https://github.com/abbbi/virtnbdbackup/discussions/48

**Comment by rettw78** [comment, marked as answer]
Thanks - that is what I needed to know have to tell libvirt the checkpoints are missing.  Interesting thing is that this domain had been migrated, but several others had as well, but this is the only one to exhibit the issue.  

Anyway, thanks again - I know have a full backup running for the domain in question.

Rett Walters

**Comment by abbbi** [comment]
reopen, so its easier for other users to find.
